src/UserSpaceInstrumentation/include/UserSpaceInstrumentation/RegisterState.h

```diff
@@ -0,0 +1,240 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef USER_SPACE_INSTRUMENTATION_REGISTER_STATE_H_
+#define USER_SPACE_INSTRUMENTATION_REGISTER_STATE_H_
+
+#include <stdint.h>
+
+#include <array>
+#include <vector>
+
+#include "OrbitBase/Result.h"
+
+namespace orbit_user_space_instrumentation {
+
+// The structs below are defined to match the memory layout of the binary blobs containing the
+// register state of the cpu (as returned by the kernel). They should not used directly - see
+// comment on class "RegisterState" below.
+#define PACKED_START _Pragma("pack(push, 1)")
+#define PACKED_END _Pragma("pack(pop)")
+
+PACKED_START
+struct GeneralPurposeRegisters32 {
+  uint32_t ebx;
+  uint32_t ecx;
+  uint32_t edx;
+  uint32_t esi;
+  uint32_t edi;
+  uint32_t ebp;
+  uint32_t eax;
+  uint32_t xds;
+  uint32_t xes;
+  uint32_t xfs;
+  uint32_t xgs;
+  uint32_t orig_eax;
+  uint32_t eip;
+  uint32_t xcs;
+  uint32_t eflags;
+  uint32_t esp;
+  uint32_t xss;
+};
+
+struct GeneralPurposeRegisters64 {
+  uint64_t r15;
+  uint64_t r14;
+  uint64_t r13;
+  uint64_t r12;
+  uint64_t rbp;
+  uint64_t rbx;
+  uint64_t r11;
+  uint64_t r10;
+  uint64_t r9;
+  uint64_t r8;
+  uint64_t rax;
+  uint64_t rcx;
+  uint64_t rdx;
+  uint64_t rsi;
+  uint64_t rdi;
+  uint64_t orig_rax;
+  uint64_t rip;
+  uint64_t cs;
+  uint64_t rflags;
+  uint64_t rsp;
+  uint64_t ss;
+  uint64_t fs_base;
+  uint64_t gs_base;
+  uint64_t ds;
+  uint64_t es;
+  uint64_t fs;
+  uint64_t gs;
+};
+
+union GeneralPurposeRegisters {
+  GeneralPurposeRegisters32 x86_32;
+  GeneralPurposeRegisters64 x86_64;
+};
+
+struct MmsAs80BitFloat {
+  uint64_t mantissa;
+  uint16_t sign_exp;
+};
+static_assert(sizeof(MmsAs80BitFloat) == 10, "MmsAsFloat is not 10 bytes of size");
+
+struct MmsRegister {
+  union {
+    std::array<uint8_t, 10> bytes;
+    MmsAs80BitFloat as_float;
+  };
+  std::array<uint8_t, 6> reserved;
+};
+static_assert(sizeof(MmsRegister) == 16, "MmsRegister is not 16 bytes of size");
+
+struct XmmRegister {
+  uint8_t bytes[16];
```

Comment:
Is there a reason why you use `std::array` above but a "normal" array here?

Reply:
no, just forget it. done.